pkg/middleware/middleware.go

```diff
@@ -72,6 +74,28 @@ func Process(ctx *quaycontext.QuayRegistryContext, quay *v1.QuayRegistry, obj cl
 		return dep, nil
 	}
 
+	// If the current object is a PVC, check for volume override
+	if pvc, ok := obj.(*corev1.PersistentVolumeClaim); ok {
+		var volumeSizeOverride resource.Quantity
+		switch quayComponentLabel {
+		case "postgres":
+			volumeSizeOverride, err = getVolumeSizeOverrideForComponent(quay, v1.ComponentPostgres)
+			if err != nil {
+				return nil, fmt.Errorf("could not override volume size: " + err.Error())
+			}
+		case "clair-postgres":
+			volumeSizeOverride, err = getVolumeSizeOverrideForComponent(quay, v1.ComponentClair)
+			if err != nil {
+				return nil, fmt.Errorf("could not override volume size: " + err.Error())
+			}
+		}
+		if volumeSizeOverride.Cmp(*pvc.Spec.Resources.Requests.Storage()) == -1 {
+			return nil, fmt.Errorf("cannot shrink volume override size from %s to %s", volumeSizeOverride.String(), pvc.Spec.Resources.Requests.Storage().String())
```

Comment:
We shouldn't blindly trust that we can dereference the return of `Storage()` here. Another thing is that this makes us not to support volume shrink, I guess this is OK but is the Jira card reporter aware of this? Can you discuss this and keep the card updated ? (makes QE and our life easier down the line).

I have noticed that storage resizes for mounted volumes only occur when the pod is restarted (at least for gp2 storage classes), remaining with a `FileSystemResizePending` condition where users can read: 

```Waiting for user to (re-)start a pod to finish file system resize of volume on node.```

It remains in this state until the pod where it is mounted is restarted. How are we restarting postgres pods ? Shouldn't we take this condition into account when reporting statuses for `postgres` and `clair` components?

Reply:
I've added a nil check for the Storage() value just in case, however in this scenario it would never be nil anyway. Also, we cannot support volume shrinking as this is not even supported by most, if not all, storage classes. Also, the PVCs are created during the re-reconcile (which would be triggered by setting an override). The pods are also redeployed in that reconcile loop and will mount the new volume.